File: ui/base/src/main/java/org/cru/godtools/base/ui/fragment/BaseFragment.java
package org.cru.godtools.base.ui.fragment;

import android.os.Bundle;
import android.view.View;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;
import butterknife.ButterKnife;
import butterknife.Unbinder;

public abstract class BaseFragment extends Fragment {
    @Nullable
    private Unbinder mButterKnife;

    // region Lifecycle Events

    @Override
    public void onViewCreated(@NonNull final View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);
        mButterKnife = ButterKnife.bind(this, view);
    }

    @Override
    public void onDestroyView() {
        if (mButterKnife != null) {
            mButterKnife.unbind();
        }
        mButterKnife = null;
        super.onDestroyView();
    }

    // endregion Lifecycle Events
}

File: ui/base/src/main/kotlin/org/cru/godtools/base/ui/fragment/BaseFragment.kt
package org.cru.godtools.base.ui.fragment

import androidx.annotation.LayoutRes
import androidx.viewbinding.ViewBinding
import org.ccci.gto.android.common.androidx.fragment.app.BindingFragment

abstract class BaseFragment<B : ViewBinding>(@LayoutRes contentLayoutId: Int) : BindingFragment<B>(contentLayoutId)
